trade-finance-manager-ui/server/controllers/utilisation-reports/record-corrections/create-record-correction-request/index.test.ts

```diff
@@ -42,25 +47,179 @@ describe('controllers/utilisation-reports/record-corrections/create-record-corre
       exporter: 'Sample Company Ltd',
     };
     jest.mocked(api.getFeeRecord).mockResolvedValue(feeRecordResponse);
+  });
 
-    // Act
-    await createRecordCorrectionRequest(req, res);
-
-    // Assert
-    expect(res._getRenderView()).toEqual('utilisation-reports/record-corrections/create-record-correction-request.njk');
-    expect(res._getRenderData() as CreateRecordCorrectionRequestViewModel).toEqual<CreateRecordCorrectionRequestViewModel>({
-      bank: { name: 'Test Bank' },
-      user,
-      activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.UTILISATION_REPORTS,
-      reportId,
-      formattedReportPeriod: 'January 2024',
-      feeRecord: {
-        facilityId: '0012345678',
-        exporter: 'Sample Company Ltd',
-      },
-      backLinkHref: getLinkToPremiumPaymentsTab(reportId, [456]),
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('getCreateRecordCorrectionRequest', () => {
+    const getHttpMocks = () =>
+      httpMocks.createMocks<GetCreateRecordCorrectionRequestRequest>({
+        params: { reportId, feeRecordId },
+        session: requestSession,
+      });
+
+    it('should render the create record correction request page', async () => {
+      // Arrange
+      const { req, res } = getHttpMocks();
+
+      // Act
+      await getCreateRecordCorrectionRequest(req, res);
+
+      // Assert
+      expect(res._getRenderView()).toEqual('utilisation-reports/record-corrections/create-record-correction-request.njk');
+      expect(res._getRenderData() as CreateRecordCorrectionRequestViewModel).toEqual<CreateRecordCorrectionRequestViewModel>({
+        bank: { name: 'Test Bank' },
+        user,
+        activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.UTILISATION_REPORTS,
+        reportId,
+        formattedReportPeriod: 'January 2024',
+        feeRecord: {
+          facilityId: '0012345678',
+          exporter: 'Sample Company Ltd',
+        },
+        formValues: {},
+        errors: { errorSummary: [] },
+        backLinkHref: getLinkToPremiumPaymentsTab(reportId, [456]),
+      });
+
+      expect(api.getFeeRecord).toHaveBeenCalledTimes(1);
+      expect(api.getFeeRecord).toHaveBeenCalledWith(reportId, feeRecordId, userToken);
```

Comment:
Can we extract the `api.getFeeRecord` into their own test please

Reply:
Pulled these out into a unit test for both endpoints (get and post) - please let me know if this isn't in line with what you were intending thanks